src/commands/register.ts

```diff
@@ -2,142 +2,155 @@ import { flags } from '@oclif/command';
-          try {
-            await pushImage(image);
-          } catch (err) {
-            cli.action.stop(chalk.red(`Push failed for image ${image}`));
-            throw new Error(err);
-          }
-          cli.action.stop(chalk.green(`Successfully pushed Docker image for ${image}`));
-        }
+    for (const config_path of config_paths) {
+      await this.register_component(config_path, flags.tag);
+    }
+  }
 
-        const [account_name, _] = node.service_config.getName().split('/');
-        const selected_account = this.accounts.rows.find((a: any) => a.name === account_name);
-        if (!selected_account) {
-          throw new Error(`You do not have access to the account specified in your service config: ${account_name}`);
-        }
+  private async register_component(config_path: string, tag: string) {
+    const component_config = await ComponentConfigBuilder.buildFromPath(config_path);
 
-        cli.action.start(chalk.blue(`Running \`docker inspect\` on the given image: ${image}`));
-        const digest = await getDigest(image).catch(e => {
-          cli.action.stop(chalk.red(`Inspect failed`));
-          throw new Error(`The image specified in your ServiceConfig is not reachable by docker: ${node.service_config.getImage()}`);
-        });
-        cli.action.stop(chalk.green(`Image verified`));
-
-        const image_without_tag = strip_tag_from_image(image); // we don't need the tag on our image because we use the digest as the key.
-
-        const service_dto = {
-          tag: flags.tag,
-          digest: digest,
-          image: image_without_tag,
-          config: classToPlain(node.service_config),
-        };
-        cli.action.start(chalk.blue(`Registering service ${node.service_config.getName()}:${flags.tag} with Architect Cloud...`));
-        await this.post_service_to_api(service_dto, selected_account.id);
-        cli.action.stop(chalk.green(`Successfully registered service`));
-      }
+    const account_name = component_config.getRef().split('/')[0];
+    const selected_account = this.accounts.rows.find((a: any) => a.name === account_name);
+    if (!selected_account) {
+      throw new Error(`You do not have access to the account specified in your component config: ${account_name}`);
+    }
+
+    for (const [service_ref, service_config] of Object.entries(component_config.getServices())) {
+      const image_tag = `${this.app.config.registry_host}/${component_config.getName()}-${service_config.getName()}:${tag}`;
+      const image = await this.push_image_if_necessary(config_path, service_config, image_tag);
+      component_config.getServiceByRef(service_ref)?.setImage(image);
     }
+
+    const component_dto = {
+      tag: tag,
+      config: classToPlain(component_config),
+    };
+
+    cli.action.start(chalk.blue(`Registering component ${component_config.getName()}:${tag} with Architect Cloud...`));
+    await this.app.api.post(`/accounts/${selected_account.id}/components`, component_dto);
```

Comment:
We can talk after standup, but this will post the transformed config - ideally we would store the unaltered config.

I'll merge for now, but we should address this.

Reply:
For this: do you think we need a `loadUnmodifiedConfigFromPath(config_path)` as part of the `ComponentConfigBuilder`? Because basically what I'm hearing is we just want to do a yml parse or json parse and send as is correct?

Reply:
you're hearing correct :/ 

I already added a `readFromPath` to the componentconfigbuilder